### PR TITLE
Fixed altitude in interpolate filter

### PR DESCRIPTION
**Pull request by cj4567**
The interpolate filter currently interpolates only longitude and latitude. Fixed it so it also handles altitude.

### CODE REVIEW

File: reference/track/tinterptrack.gpx [hunk 1]
```diff
@@ -1,53 +1,53 @@
...
-        <time>2002-05-25T17:06:09Z</time>
-      </trkpt>
-      <trkpt lat="36.000000000" lon="-115.000000000">
-        <ele>0.000000</ele>
-        <time>2002-05-25T17:06:10Z</time>
-      </trkpt>
-    </trkseg>
-  </trk>
-</gpx>
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (GPSBabelDeveloper) [inline reference/track/tinterptrack.gpx]: No carriage returns in the repo, please.

File: reference/track/interptrack.gpx [hunk 2]
```diff
@@ -1,149 +1,149 @@
...
-        <time>2002-05-25T17:06:09Z</time>
-      </trkpt>
-      <trkpt lat="36.000000000" lon="-115.000000000">
-        <ele>0.000000</ele>
-        <time>2002-05-25T17:06:10Z</time>
-      </trkpt>
-    </trkseg>
-  </trk>
-</gpx>
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (GPSBabelDeveloper) [inline reference/track/interptrack.gpx]: No carriage returns in the repo, please.

File: interpolate.cc [hunk 3]
```diff
@@ -108,12 +110,14 @@ interpfilt_process(void)
             wpt_new->shortname = QString();
             wpt_new->description = QString();
 
+            frac = (double)(timen - time1) / (double)(wpt->creation_time.toTime_t() - time1);
```
Comment (GPSBabelDeveloper) [inline interpolate.cc:113]: This probably breaks down for time < 1970 or > 2038 or something)  This problem isn't new and it's not your fault. Anyplace that a double is used to hold a time is probably slightly wrong these days. You could probably make frac a QDateTime and eliminate the toTime_t() and rely on the overloaded +/- operators to DTRT. If that doesn't fall out within a couple minutes, that's cool; I won't push back more than a bit on that.

**Review by GPSBabelDeveloper — CHANGES_REQUESTED** [review]
Nice! Shows a good understanding of the problem. Thanx. If changing that to a QDateTime doesn't fall out, I'll take it after the CR's are removed.

File: interpolate.cc (hunk 3 above)
Comment (cj4567) [reply interpolate.cc:113]: frac is just between 0 and 1 here and does not hold the time. You're right though, toTime_t() will break before 1970 and after 2106. Replaced it with toSecsSinceEpoch().

File: reference/track/interptrack.gpx (hunk 2 above)
Comment (cj4567) [reply reference/track/interptrack.gpx]: Sorry, fixed it.

File: interpolate.cc (hunk 3 above)
Comment (cj4567) [reply interpolate.cc:113]: Never mind, that's only available in newer qt versions...

**Review by GPSBabelDeveloper — APPROVED** [review]
That's due diligence. Thanx for looking into it.

**Pull request merged**